### PR TITLE
Setting up the package internals

### PR DESCRIPTION
**Pull request by jbusecke**
This is very much a work in progress rn. 

I am trying to transfer @rabernat s work to a clean repo, and eventually make it pip and conda installable.

### CODE REVIEW

**Comment by rabernat** [comment]
I would but the submodule inside a subdirectory called src. In src, we will also include the wrapper fortran file and the signature file. So
```
src/wrapper.f90
src/wrapper.pyf
src/aerobulk/...
```

**Comment by jbusecke** [comment]
Ok I can do that.

How do you think is the best way of linking the files in `aerobulk/src`:
Softlink them into the `aerobulk-python/src` directory?

**Comment by rabernat** [comment]
No links are necessary. We just provide the appropriate paths to f2py / distutils. So something like `src/aerobulk/src/mod_*.f90`.

**Comment by jbusecke** [comment]
Ok so with the current state of this PR I was able to get the package to work manually: 
I followed [these](https://github.com/ocean-transport/scale-aware-air-sea/issues/11#issuecomment-1089201824) steps and then modified the last step (how do we call the final compile/build? for f2py?) like this:
```
python -m numpy.f2py --verbose -c --f90flags="-fdefault-real-8 -ffree-line-length-200 --std=gnu" -I./aerobulk/mod mod_aerobulk_wrap.pyf  mod_aerobulk_wrap.f90 aerobulk/src/mod_*.f90
```
in the command line. 

I am not thinking/testing how/if I could get this to work within the `setup.py` file.

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    f90flags.append("-132")
+#  Set aggressive optimization level
+f90flags.append("-O3")
+#  Suppress all compiler warnings (avoid huge CI log files)
+f90flags.append("-w")
+
+# for this API we will only expose a single extension?
+ext_modules = [
+    Extension(
+        name="mod_aerobulk_wrap",  # this somehow needs to match the mod_aerobulk_wrap.* files. Ideally I would like to name this just aerobulk (for later.)
```
Comment (rabernat) [inline setup.py:35]: Feel free to rename it however you want. This naming was just my first attempt.

Comment (jbusecke) [reply setup.py:35]: Left that to #5 for later!

**Pull request merged**